### PR TITLE
Learn trusted headers

### PR DESCRIPTION
**Pull request by samhotchkiss**
Locally, so that we're caching blocks properly

### CODE REVIEW

File: modules/protect.php [hunk 1]
```diff
@@ -84,9 +84,46 @@ public function maybe_get_protect_key() {
...
+				foreach( $headers as $header => $ip ) {
+					
+					$ips = explode( ', ', $ip );
+					
+					$ip_list_has_nonprivate_ip = false;
+					foreach( $ips as $ip ) {
+						$ip = jetpack_clean_ip( $ip );
+						
+						// If the IP is in a private or reserved range, return REMOTE_ADDR to help prevent spoofing
+						if ( $ip == '127.0.0.1' || $ip == '::1' || jetpack_protect_ip_is_private( $ip ) ) {
```
Comment (roccotripaldi) [inline modules/protect.php]: I realize the first two IPs are local hosts, but should we add them to the jetpack_protect_ip_is_private() method?

**Comment by roccotripaldi** [comment]
This seems solid. Are we still cacheing the trusted headers on the API server as well?

**Comment by samhotchkiss** [comment]
@roccotripaldi - yep, this is just to improve caching performance

File: modules/protect.php (hunk 1 above)
Comment (samhotchkiss) [inline modules/protect.php]: The reason that we'd split those out previously was that they are the most common, so we want to allow them to get caught early for performance reasons.

**Comment by dereksmart** [comment]
nice, looks good :+1:

**Pull request merged**